src/scheduler/gc_work.rs

```diff
@@ -599,7 +599,7 @@ pub trait ProcessEdgesWork:
             // Executing these work packets now can remarkably reduce the global synchronization time.
             self.worker().do_work(work_packet);
         } else {
-            self.mmtk.scheduler.work_buckets[WorkBucketStage::Closure].add(work_packet);
+            self.mmtk.scheduler.work_buckets[&WorkBucketStage::Closure].add(work_packet);
```

Comment:
If this `&` occurs frequently, we may consider this as a chance to turn the `work_buckets` vec into a private field, and force all accesses through a method such as `scheduler.add_work_packet(stage, packet);`  We can let the `stage` argument be `impl Into<WorkBucketStage>` so that we can pass `PlanSpecificStage` instances directly.